### PR TITLE
chore: upgrade jest to ^29

### PR DESCRIPTION
**Pull request by corymhall**
Now that we are on typescript >4.2 we can upgrade to jest 29.

- Updated some snapshots because the snapshot format in jest 29 changed.
- Cleaned up some test output by spying logs

All other updates were just as a result of the version update

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/core/test/validation/validation.test.ts [hunk 1]
```diff
@@ -50,6 +53,8 @@ describe('validations', () => {
     app.synth();
     expect(process.exitCode).toEqual(1);
 
+    // eslint-disable-next-line no-console
```
Comment (vinayak-kukreja) [inline packages/aws-cdk-lib/core/test/validation/validation.test.ts:56]: I believe this can be removed :)

**Review by vinayak-kukreja — CHANGES_REQUESTED** [review]

File: packages/aws-cdk-lib/core/test/validation/validation.test.ts (hunk 1 above)
Comment (corymhall) [inline packages/aws-cdk-lib/core/test/validation/validation.test.ts:56]:
```suggestion
```

Comment (corymhall) [reply packages/aws-cdk-lib/core/test/validation/validation.test.ts]: Good catch! Updated.

File: packages/aws-cdk-lib/core/test/validation/validation.test.ts [hunk 2]
```diff
@@ -50,6 +53,7 @@ describe('validations', () => {
     app.synth();
     expect(process.exitCode).toEqual(1);
 
+    console.log(consoleErrorMock.mock.calls[0][0]);
```
Comment (corymhall) [inline packages/aws-cdk-lib/core/test/validation/validation.test.ts:56]:
```suggestion
```

**Review by vinayak-kukreja — APPROVED** [review]

**Pull request merged**